Pulsarc/Skinning/Skin.cs

```diff
@@ -238,7 +256,75 @@ static public float getConfigFloat(string config, string section, string key)
         /// <returns>The Anchor found using the provided parameters.</returns>
         static public Anchor getConfigAnchor(string config, string section, string key)
         {
-            return (Anchor) Enum.Parse(Anchor.TopLeft.GetType(),configs[config][section][key]);
+            return (Anchor)Enum.Parse(Anchor.TopLeft.GetType(), getConfigString(config, section, key));
+        }
+
+        /// <summary>
+        /// Find the config provided, go to the section provided, and return the start position of the
+        /// key provided. If parent is not null, the start position will be based on the parent.
+        /// If parent is null, start position will be based on the screen.
+        /// </summary>
+        /// <param name="config">The config to look in.</param>
+        /// <param name="section">The section of a config to look in.</param>
+        /// <param name="key">The name of the variable.</param>
+        /// <returns>
+        /// The position found using the provided parameters.
+        /// Currently only ScreenAnchor and pre-determined parent position finding is supported.
+        /// </returns>
+        static public Vector2 getConfigStartPosition(string config, string section, string key, Drawable parent = null)
+        {
+            string anchorString = getConfigString(config, section, key);
+
+            if (parent == null)
+            {
+                return AnchorUtil.FindScreenPosition((Anchor)Enum.Parse(Anchor.TopLeft.GetType(), anchorString));
+            }
+            else
+            {
+                return AnchorUtil.FindDrawablePosition((Anchor)Enum.Parse(Anchor.TopLeft.GetType(), anchorString), parent);
+
+            }
+        }
+
+        /// <summary>
+        /// Find the config provided, go to the section provided, and return the Color of the
+        /// key provided. Format like this: "r, g, b, [a]"
+        /// </summary>
+        /// <param name="config">The config to look in.</param>
+        /// <param name="section">The section of a config to look in.</param>
+        /// <param name="key">The name of the variable.</param>
+        /// <returns>The Color found using the provided parameters.</returns>
+        static public Color getConfigColor(string config, string section, string key)
+        {
+            string valueToParse = getConfigString(config, section, key);
+
+            var parts = valueToParse.Split(',');
+            int r, g, b = 0;
+            int a = 255;
+
+            try
+            {
+                r = int.Parse(parts[0]);
+                g = int.Parse(parts[1]);
+                b = int.Parse(parts[2]);
+
+                if (parts.Length > 3)
+                {
+                    a = int.Parse(parts[3]);
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.Error(key + " was not formatted correctly." +
```

Comment:
Perhaps use string interpolation here?
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

Reply:
Yes, i learned about these recently so i didn't use it in my code, we should use that for all these cases

Reply:
Fixed, much easier to read now, thanks! I'll keep this in mind while cleaning up other strings in the code.